src/universal/containers/app/TemplateDetailPage/TemplateDetailOverview/TemplateDetailOverview.js

```diff
@@ -67,64 +76,82 @@ export default class TemplateDetailOverview extends Component {
-              <RaisedButton
-                labelText="Commit to production"
-                onClick={::this.showCommitToProductionForm}
-                primary />
+        <div className={style.TemplateDetailOverview_content}>
+          <Paper className={style.TemplateDetailOverview_actionButtonGroup}>
+            <div>
+              <div>Choose a layout:</div>
+              <SelectField menuItems={this.layoutsSelectItems}/>
             </div>
 
-            <div className={style.TemplateDetailOverview_actionButtonContainer}>
-              <RaisedButton
-                labelText="Preview html"
-                secondary />
+            <div>
+              <div>Choose a language:</div>
+              <SelectField menuItems={this.languagesSelectItems}/>
             </div>
 
-            <div className={style.TemplateDetailOverview_actionButtonContainer}>
-              <RaisedButton
-                labelText="Preview pdf"
-                secondary />
+            <div>
+              <div className={style.TemplateDetailOverview_actionButtonContainer}>
+                <RaisedButton
+                  labelText="Preview html"
+                  secondary />
+              </div>
+              <div className={style.TemplateDetailOverview_actionButtonContainer}>
+                <RaisedButton
+                  labelText="Preview pdf"
+                  secondary />
+              </div>
+              <div className="clearfix"/>
+              {!isReadOnly && <div className={style.TemplateDetailOverview_commitButtonContainer}>
+                <RaisedButton
+                  labelText="Commit to production"
+                  onClick={::this.showCommitToProductionForm}
+                  primary />
+              </div>}
+              <div className="clearfix"/>
             </div>
-            <div className="clearfix"/>
-          </div>
-        </Paper>
-
-        <Paper className={style.TemplateDetailOverview_versionsContainer}>
-          <div>
-            <h3>Template versions</h3>
-            {versions.map(this.renderTemplateVersion)}
-          </div>
-        </Paper>
-
-        <br />
+          </Paper>
+
+          <Paper className={style.TemplateDetailOverview_versionsContainer}>
+            <div>
+              <TemplateVersionsList
+                subheader="Template Versions"
+                items={versions}
+                onViewPressed={(item) => {
+                  const {objectId} = item;
+                  this.props.loadVersionHistory(objectId);
+                }}
+                onSetProductionPressed={() => {}}
+                primaryText={(item) => {
+                  return <span>{item.commitMessage}</span>;
+                }}
+                secondaryText={(item) => {
+                  const text = moment(item.createdAt).calendar();
+                  const viewingSpan = <span className={style.TemplateDetailOverview__isViewingVersion}>(viewing)</span>;
```

Comment:
I'd rename this to something like viewedVersionMaker. The "span" in the name tells me nothing :).